repository/api/validators/dict_validator.py

```diff
@@ -0,0 +1,34 @@
+from rest_framework import serializers
+from typing import Any, Dict, List, OrderedDict, Union
+
+
+class DictValidator:
+    """
+    TODO: description
+    """
+
+    def __init__(self, fields: List[str], nullable=True):
+        self.fields = fields
+        self.nullable = nullable
+
+    def __call__(self, attrs: OrderedDict[str, Any]):
+        error_messages = {}
+
+        for field in self.fields:
+            error_message = self.validate(field, attrs[field])
+            if error_message is not None:
+                error_messages.update(error_message)
+
+        if error_messages:
+            raise serializers.ValidationError(error_messages)
+
+    def validate(self, field: str, value: Any) -> Union[Dict[str, str], None]:
+        if value is None:
+            if not self.nullable:
+                return {f"{field}": "This field may not be null."}
+        else:
+            # Using `type` instead of `isinstance` to validate that it is a dict and no a subtype
```

Comment:
Here @IceKhan13 I used `type` instead of `isinstance` to check that we are receiving a `dict` and not an `OrderedDict`, for example. I think is correct but let me know what you think, I had some doubts.